developer_manual/app/tutorial.rst

```diff
@@ -0,0 +1,1021 @@
+
+    ocdev setup core --dir owncloud  --branch stable8
+
+First you want to enable debug mode to get proper error messages. To do that add **DEFINE('DEBUG', true);** at the end of the **owncloud/config/config.php** file::
+
+    echo "\nDEFINE('DEBUG', true);" >> owncloud/config/config.php
+
+.. note:: PHP errors are logged to **owncloud/data/owncloud.log**
+
+Now open another terminal window and start the development server::
+
+    cd owncloud
+    php -S localhost:8080
+
+Afterwards the app can be created in the **apps** folder::
+
+    cd apps
+    ocdev startapp OwnNotes
+
+This creates a new folder called **ownnotes**. Now access and set up ownCloud through the webinterface at `http://localhost:8080 <http://localhost:8080>`_ and enable the OwnNotes application on the `apps page <http://localhost:8080/index.php/settings/apps>`_.
+
+The first basic app is now available at `http://localhost:8080/index.php/apps/ownnotes/ <http://localhost:8080/index.php/apps/ownnotes/>`_
+
+Routes & Controllers
+====================
+A typical web application consists of server side and client side code. The glue between those two parts are the URLs. In case of the notes app the following URLs will be used:
+
+* **GET /**: Returns the interface in HTML
+* **GET /notes**: Returns a list of all notes in JSON
+* **GET /notes/1**: Returns a note with the id 1 in JSON
+* **DELETE /notes/1**: Deletes a note with the id 1
+* **POST /notes**: Creates a new note by passing in JSON
+* **PUT /notes/1**: Updates a note with the id 1 by passing in JSON
+
+On the client side we can call these URLs with the following jQuery code:
+
+.. code-block:: js
+
+    // example for calling the PUT /notes/1 URL
+    var baseUrl = OC.generateUrl('/apps/ownnotes');
+    var note = {
+        title: 'New note',
+        content: 'This is the note text'
+    };
+    var id = 1;
+    $.ajax({
+        url: baseUrl + '/notes/' + id,
+        type: 'PUT',
+        contentType: 'application/json',
+        data: JSON.stringify(note)
+    }).done(function (response) {
+        // handle success
+    }).fail(function (response, code) {
+        // handle failure
+    });
+
+On the server side we need to register a callback that is executed once the request comes in. The callback itself will be a method on a :doc:`controller <controllers>` and the controller will be connected to the URL with a :doc:`route <routes>`. The controller and route for the page are already set up in **ownnotes/appinfo/routes.php**:
+
+.. code-block:: php
+
+    <?php
+    return ['routes' => [
+        ['name' => 'page#index', 'url' => '/', 'verb' => 'GET']
+    ]];
+
+This route calls the controller **OCA\\OwnNotes\\PageController->index()** method which is defined in **ownnotes/controller/pagecontroller.php**. The controller returns a :doc:`template <templates>`, in this case **ownnotes/templates/main.php**:
+
+.. note:: the @Stuff turns off security checks, see :doc:`controllers`
```

Comment:
Recommend not using @Stuff; use the real annotation names.  Might make sense to put the note after the code too.